application-chartjs-ui/src/main/resources/XWiki/ChartJS/WebHome.xml

```diff
@@ -319,13 +351,23 @@ require(['jquery', 'chart', 'datalabels', 'palette'], function($, Chart, datalab
   $('canvas.chart').each(function() {
     var canvas = $(this);
     var type = canvas.attr('data-type');
-    var options = {};
-    try {
-      options = JSON.parse(canvas.attr('data-options') || '{}');
-    } catch (exception) {
+
+    // Converts a JSON attribute value to a JavaScript object.
+    function jsonAttributeToObject(attributeName) {
```

Comment:
I don't see the value of this refactoring. You moved the code in an inner function that depends on the outer scope and that is called only once below. I feel it complicates the code.